src/helpers/Interfaces.ts

```diff
@@ -0,0 +1,107 @@
+export interface EndpointsConfig {
+  [property: string]: string;
+}
+
+export interface LocationSearch {
+  locations: {
+    id: number;
+    name: string;
+    country: string;
+    timezone: string;
+    adminArea: string;
+    lon: number;
+    lat: number;
+  }[];
+}
+
+export interface LocationInfo {
+  id: number;
+  name: string;
+  country: string;
+  timezone: string;
+  adminArea: string;
+  lon: number;
+  lat: number;
+}
+
+export interface CurrentData {
+  current: {
+    time: string;
+    symbol: string;
+    symbolPhrase: string;
+    temperature: number;
+    feelsLikeTemp: number;
+    relHumidity: number;
+    dewPoint: number;
+    windSpeed: number;
+    windDir: number;
+    windDirString: string;
+    windGust: number;
+    precipProb: number;
+    precipRate: number;
+    cloudiness: number;
+    thunderProb: number;
+    uvIndex: number;
+    pressure: number;
+    visibility: number;
+  };
+}
+
+export interface DailyForecastData {
+  forecast: {
+    date: string;
+    symbol: string;
+    symbolPhrase: string;
+    maxTemp: number;
+    minTemp: number;
+    maxFeelsLikeTemp: number;
+    minFeelsLikeTemp: number;
+    maxRelHumidity: number;
+    minRelHumidity: number;
+    maxDewPoint: number;
+    minDewPoint: number;
+    precipAccum: number;
+    snowAccum: number;
+    maxWindSpeed: number;
+    windDir: number;
+    maxWindGust: number;
+    precipProb: number;
+    cloudiness: number;
+    sunrise: string;
+    sunset: string;
+    sunriseEpoch: number;
+    sunsetEpoch: number;
+    moonrise: string;
+    moonset: string;
+    moonPhase: number;
+    uvIndex: number;
+    minVisibility: number;
+    pressure: number;
+    confidence: string;
+  }[];
+}
+
+export interface ThreeHourlyData {
+  forecast: {
+    time: string;
+    symbol: string;
+    symbolPhrase: string;
+    temperature: number;
+    feelsLikeTemp: number;
+    windSpeed: number;
+    windGust: number;
+    relHumidity: number;
+    dewPoint: number;
+    windDir: number;
+    windDirString: string;
+    precipProb: number;
+    precipAccum: number;
+    snowAccum: number;
+    cloudiness: number;
+    thunderProb: number;
+    uvIndex: number;
+    pressure: number;
+    visibility: number;
+    precipType: string;
```

Comment:
CurrentData and ThreeHourlyData have many common props, please create additional type/interface to collect them and reuse

Reply:
to avoid duplicate code
and generally CurrentData and ThreeHourlyData are the same: both of them are weather info, so it would be great to make common interface describing weather info

Reply:
```
interface WeatherData {
  time: string;
  symbol: string;
  symbolPhrase: string;
  temperature: number;
  feelsLikeTemp: number;
  windSpeed: number;
  windGust: number;
  relHumidity: number;
  dewPoint: number;
  windDir: number;
  windDirString: string;
  precipProb: number;
  cloudiness: number;
  thunderProb: number;
  uvIndex: number;
  pressure: number;
  visibility: number;
  precipType?: string;
  precipAccum?: number;
  snowAccum?: number;
}
export interface CurrentData {
  current: WeatherData;
}
export interface ThreeHourlyData {
  forecast: WeatherData[];
}
```